File: src/Valid Palindrome/Valid Palindrome.cpp
class Solution {
	public:
		bool isPalindrome(string s) {
			string real = "";
			for (int i = 0 ; i < s.length() ; i++)
			{
				char c = s[i];
				if(isdigit(c) || isalpha(c) )
				{
					if(isalpha(c)) c = tolower(c);
					real += c;
				}
			}

			int len = real.length();
			if(!len) return true;

			int pos1 = 0, pos2 = len-1, end = len>>1;
			while(true)
			{
				if( pos1 == end ) return true;
				if(real[pos1] == real[pos2])
				{
					pos1++;
					pos2--;
				}
				else return false;
			}
		}
};

File: src/Valid Palindrome/Valid Palindrome.py
class Solution:
	# @param s, a string
	# @return a boolean
	def isPalindrome(self, s):
		real = ""
		for c in s:
			if c.isalpha():
				real += c.lower()
			elif c.isdigit():
				real += c
		return real==real[::-1]
